js/recipe_manager.js

```diff
@@ -1,84 +1,119 @@
+const {
+  validateRecipeKey,
+  validateRecipe,
+  isValidKey
+} = require('../js/validation')
+
+const uuidv1 = require('uuid/v1')
+
 window.db = {
   recipes: [{
-    "id": 1,
-    "title": "Spaghetti Carbonara",
-    "ingredients": [
-      "200g pancetta or thick cut bacon, diced",
-      "4 yolks",
+    'id': 1,
+    'title': 'Spaghetti Carbonara',
+    'ingredients': [
+      '200g pancetta or thick cut bacon, diced',
+      '4 yolks'
     ],
-    "instructions": "1 Heat pasta water. Prepare rest of ingredients. Voila.",
-    "total_calories": 2600,
-    "number_of_portions": 4,
-    "calories_per_portion": 650
+    'instructions': '1 Heat pasta water. Prepare rest of ingredients. Voila.',
+    'total_calories': 2600,
+    'number_of_portions': 4,
+    'calories_per_portion': 650
   },
-  { 
-    "id": 2,
-    "title": "Toast",
-    "ingredients": [
-      "bread",
-      "cheese"
+  {
+    'id': 2,
+    'title': 'Toast',
+    'ingredients': [
+      'bread',
+      'cheese'
     ],
-    "instructions": "Put cheese beetwen bread slices. Put in the toaster. Voila",
-    "total_calories": 800,
-    "number_of_portions": 2,
-    "calories_per_portion": 400
+    'instructions': 'Put cheese beetwen bread slices. Put in the toaster. Voila',
+    'total_calories': 800,
+    'number_of_portions': 2,
+    'calories_per_portion': 400
   }
   ]
 }
 
-
 // ----------------- API -------------------
 
-const validateRecipeKey = function(key, newValue) {
-  // total_calories must be a number
-  // number_of_portions must be a number
-};
-
-const validateRecipe = function(recipe) {
-  // loop over pairs: key-value, call the function validateRecipeKey() with key, value
-};
-
 // CRUD
 
 /** Returns all recipes in the database */
-const getRecipes = function() {};
+const getRecipes = function () {
+  return window.db.recipes
+}
 
 /** Returns recipe with the given id */
-const getRecipe = function(id) {};
+const getRecipe = function (id) {
+  return window.db.recipes.find(function (el) {
+    return el.id === id
+  })
+}
 
 /** Adds new recipe to the database. Returns the new recipe's ID */
-const createRecipe = function(recipe) {
-  validateRecipe(recipe);
-  const numOfPort = recipe.number_of_portions;
-  const totalCal = recipe.total_calories;
- 
-  // TODO: calculate calories per portion;
-  // TODO: generate new ID;
+const createRecipe = function (recipe) {
+  if (!validateRecipe(recipe)) {
+    return false
+  };
+
+  const numOfPort = recipe.number_of_portions
+  const totalCal = recipe.total_calories
+  const caloriesPerPortion = totalCal / numOfPort
+  const newId = uuidv1()
```

Comment:
Semantically, this is not just some new id - it's `recipeId`, right?

Reply:
newRecipeId is my updated name

Reply:
I will change the name for newRecipeId - is it ok?

Reply:
We have talk offline and we've decided to leave it as is.